### ISSUE TITLE
When connecting to a ms sql db, loading tables takes a long time

### ISSUE THREAD

**Issue by joshferrell** [sqlectron/sqlectron-gui#263]
I'm currently trying to connect to one my ms sql servers. This server has a bunch of databases with a lot of tables. I'm unable to expand any of the databases, and the bottom of the app just flashes "loading list of tables."

My theory is that because this server has a lot of tables, it's trying to load **all** of the tables at once in the server. If this is the case, this should be looked into being reworked as it would probably be best that on expansion of a database, load its tables.

**Comment by maxcnunes** [comment]
Nowadays when you connect to a server, you connect to a specific database. Once you are connected to that server then you are able to connect to the other databases in that server. The tables for each database are only fetched once you connect to that database. If the database your are connecting has a really large number of tables than it can really be a performance problem. Could you confirm how many tables it has? There is a PR https://github.com/sqlectron/sqlectron-gui/pull/200 to improve the performance in the sidebar. But it has never been finished :(

**Comment by joshferrell** [comment]
Sorry for the late response,

I'm not able to see how many tables it has, but I am selecting a specific database to login to as part of my connection. However, instead of logging into a specific database, it logs into all tables that user has access to.

**Comment by maxcnunes** [comment]
Actually it logs to a single DB. It only shows the other databases available in the sidebar. But you are not connected to them. I have changed in the last version to use different database color background:

* transparent: not connected DBs
* blue: connected DBs
* yellow: current connected DB which the current query tab is related with

Seems this issue is a bit related to this one https://github.com/sqlectron/sqlectron-gui/issues/262

**Comment by joshferrell** [comment]
Ah, this makes it a lot more clear as to what's attempting to connect. 

It appears that the database I'm attempting to connect to is what's being loaded; however, that database is unable to load the list of tables. I'm able to query the table, so I know it's not a permissions issue or connection issue. It looks almost like it's infinitely refreshing the table the way the loading list of tables message flashes.

**Comment by maxcnunes** [comment]
Is this problem similar to this one? https://github.com/sqlectron/sqlectron-gui/issues/242

**Comment by maxcnunes** [comment]
@joshferrell I have added support to configure a preload filter for database and schema in the latest version. You have to configure it manually for while in the `.sqlectron.json`. That configuration should avoid this problem since you will be able to load only the database/schema that are useful for you.

**Comment by blakek** [comment]
Not sure if this is related or new issue: I seem to be having a similar issue only on MSSQL servers, and I only noticed this issue after moving from v1.18.0 to v1.19.0.

The sidebar list of tables never loads.  The "Loading list of tables..." text is flickering while it seems to try to connect to the server.  There are no errors in the developer console.  Running `select table_name from information_schema.tables` works as expected, and querying the tables works successfully.

There are approximately 500 total tables on the DB server listed.  I've tried selecting a default database, running with no default database installed, and adding filters in the `~/.sqlectron.json` file.

Example:
![demo](https://cloud.githubusercontent.com/assets/4932702/22040365/36d5f8a4-dcc8-11e6-8726-511ab43465f1.gif)
